client/pages/dashboard/dashboard.header.controller.js

```diff
@@ -1,22 +1,17 @@
-angular.module('probedock.dashboardPage').controller('DashboardHeaderCtrl', function(orgEditModal, $scope, $state, $stateParams) {
+angular.module('probedock.dashboardPage').controller('DashboardHeaderCtrl', function(orgEditModal, $scope, $state, states) {
 
-  var modal;
-  $scope.currentState = $state.current.name;
+  states.onStateChangeSuccess($scope, 'org.dashboard.default.edit', function(state, params, resolves) {
 
-  $scope.$on('$stateChangeSuccess', function(event, toState) {
+    var modal = orgEditModal.open($scope, {
+      organizationName: resolves.routeOrgName
```

Comment:
I think both would work in this case, because it seems the dashboard header controller is recreated even if you say on the dashboard and switch organizations.

But in a case where you would change state without this controller being recreated, then you would need to get it from the state's resolves and not from the original injection, which wouldn't be updated.